mantidproject/mantid
# Conventional cells for satellite peaks

**Description of work.**
Get conventional cell with modulated vectors



**To test:**
[11440_Niggli.txt](https://github.com/mantidproject/mantid/files/3591841/11440_Niggli.txt)

In the following, `HKL` and `ModUB` should now be correct
```python
from mantid.simpleapi import *

peaks=LoadIsawPeaks(Filename='/home/rwp/11440_Niggli.txt', OutputWorkspace='11440_Niggli_peaks')
IndexPeaks(PeaksWorkspace='11440_Niggli_peaks', Tolerance=0.12, RoundHKLs=False, CommonUBForAll=True)
FindUBUsingIndexedPeaks(PeaksWorkspace='11440_Niggli_peaks', Tolerance=0.12)
OptimizeLatticeForCellType(PeaksWorkspace='11440_Niggli_peaks',
    CellType='Monoclinic', Apply=True, EdgePixels=16)
FindUBUsingIndexedPeaks(PeaksWorkspace='11440_Niggli_peaks', Tolerance=0.12)

print("HKL = {}".format(peaks.getPeak(0).getHKL()))
print("IntHKL = {}".format(peaks.getPeak(0).getIntHKL()))
print("UB = {}".format(peaks.sample().getOrientedLattice().getUB()))
print("ModUB = {}".format(peaks.sample().getOrientedLattice().getModUB()))
```
Before:
```
HKL = [-3,1,0]
IntHKL = [-3,1,0]
UB = [[ 0.01079288 -0.04743855 -0.06457541]
 [ 0.12603299 -0.03387132  0.05001882]
 [-0.13934267 -0.03422954  0.03994415]]
ModUB = [[ 0.  0.  0.]
 [ 0.  0.  0.]
 [ 0.  0.  0.]]
```

Now:
```
HKL = [-2.9968,0.995042,0.00628075]
IntHKL = [-3,1,0]
UB = [[ 0.01079267 -0.04743867 -0.06457663]
 [ 0.12603283 -0.03387138  0.05001714]
 [-0.1393427  -0.03422955  0.03994408]]
ModUB = [[-0.0322847   0.          0.        ]
 [ 0.02502384  0.          0.        ]
 [ 0.01997751  0.          0.        ]]
```

Fixes #25741 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
